scripts/pipeline/populate_per_genome_database.pl

```diff
@@ -0,0 +1,291 @@
+    copy_homology_and_members( $pipeline_dba, $compara_dba, $genome_db );
+}
+
+$compara_dba->dbc->do("SET FOREIGN_KEY_CHECKS = 1");
+
+
+exit(0);
+
+=head2 copy_genome_db
+
+    Arg[1]      : Bio::EnsEMBL::Compara::DBSQL::DBAdaptor $to_dba (Mandatory)
+    Arg[2]      : Bio::EnsEMBL::Compara::DBSQL::GenomeDBAdaptor $genome_db (Mandatory)
+    Description : copy from $from_dba to $to_dba just the genome_db which
+                  correspond to $genome_db only.
+    Returns     : None
+    Exceptions  : None
+
+=cut
+
+sub copy_genome_db {
+    my ($to_dba, $genome_db) = @_;
+
+    my $genome_dba = $to_dba->get_GenomeDBAdaptor;
+    $genome_dba->store( $genome_db );
+}
+
+=head2 copy_mlss_and_ss
+
+    Arg[1]      : Bio::EnsEMBL::Compara::DBSQL::DBAdaptor $from_dba (Mandatory)
+    Arg[2]      : Bio::EnsEMBL::Compara::DBSQL::DBAdaptor $to_dba (Mandatory)
+    Arg[3]      : Bio::EnsEMBL::Compara::DBSQL::GenomeDBAdaptor $genome_db (Mandatory)
+    Description : copy from $from_dba to $to_dba just the method_link_species_set
+                  and species_sets which correspond to $genome_db_id only.
+    Returns     : None
+    Exceptions  : None
+
+=cut
+
+sub copy_mlss_and_ss {
+    my ($from_dba, $to_dba, $genome_db) = @_;
+
+    my $from_mlss_adap = $from_dba->get_MethodLinkSpeciesSetAdaptor;
+    my $to_mlss_adap   = $to_dba->get_MethodLinkSpeciesSetAdaptor;
+
+    my $mlsss = $from_mlss_adap->fetch_all_by_GenomeDB( $genome_db );
+    foreach my $mlss ( @$mlsss ) {
+        $to_mlss_adap->store( $mlss );
+    }
+}
+
+=head2 copy_gene_and_seq_members
+
+    Arg[1]      : Bio::EnsEMBL::Compara::DBSQL::DBAdaptor $from_dba (Mandatory)
+    Arg[2]      : Bio::EnsEMBL::Compara::DBSQL::DBAdaptor $to_dba (Mandatory)
+    Arg[3]      : Bio::EnsEMBL::Compara::DBSQL::GenomeDBAdaptor $genome_db (Mandatory)
+    Arg[4]      : $dnafrag default: undef (Optional)
+    Description : copy from $from_dba to $to_dba just the gene_member and
+                  seq_members which correspond to $genome_db_id only. Optionally
+                  copies across dnafrags if required.
+    Returns     : None
+    Exceptions  : None
+
+=cut
+
+sub copy_gene_and_seq_members {
+    my ($from_dba, $to_dba, $genome_db, $dnafrag) = @_;
+
+    my $from_gene_adap = $from_dba->get_GeneMemberAdaptor;
+    my $to_gene_adap   = $to_dba->get_GeneMemberAdaptor;
+
+    my $from_dna_adap = $from_dba->get_DnaFragAdaptor;
+    my $to_dna_adap   = $to_dba->get_DnaFragAdaptor;
+
+    my $from_sm_adap = $from_dba->get_SeqMemberAdaptor;
+    my $to_sm_adap   = $to_dba->get_SeqMemberAdaptor;
+
+    my $from_seq_adap = $from_dba->get_SequenceAdaptor;
+    my $to_seq_adap   = $to_dba->get_SequenceAdaptor;
+
+    my $seq_members = $from_sm_adap->fetch_all_canonical_by_GenomeDB( $genome_db );
+    foreach my $seq_member ( @$seq_members ) {
+        # For some reason $seq_member->gene_member returns undef attribute.
+        # May be linked to intentional missing dnafrag for homology etc.
```

Comment:
Based on `SeqMember.pm` code, and since `$seq_member->gene_member_id` is returning something, it looks like `$seq_member->adaptor` is not defined? Because what you are doing below is exactly what the code for `$seq_member->gene_member` does. Is this a bug in the API? Should we fix it before it affects other pipelines?

Reply:
Yes, I too looked into the code - I think it is a bug in the API - but I don't think it affects the normal release db objects. I ran out of time to look into it, so I put this here as a reminder. I will write a ticket for it - because otherwise this works for now.